### ISSUE TITLE
Use metadata-less Filetree in smalltalkCI

### ISSUE THREAD

**Issue by fniephaus** [hpi-swa/smalltalkCI#334]
This makes it hard for other users to contribute. On the other hand, I'm not sure if all images supported by smalltalkCI also support the metadata-less format. @dalehenrich?

**Comment by estebanlm** [comment]
AFAIK metadataless format is available for everybody. Why should make it harder?
(unless it is not there...)
honestly, I think is metadatafull what makes harder to contribute to a project.

**Comment by fniephaus** [comment]
The reason why smalltalkCI still uses the metadata-full format is purely to provide backwards compatibility. I need to update the Squeak images with the latest Metacello/Filetree. Are you sure `Pharo-3.0` and all other Pharo images can ready metadata-less Filetree out of the box?

**Comment by estebanlm** [comment]
mmm.... P4+ yes... P3 not sure.

**Comment by fniephaus** [comment]
If P3 and P4 are not longer needed, we can deprecate support for them. But I'm sure there are projects using them and GemStone has P3/P4 clients too, so that's probably not an option. Can we update Metacello in P3/P4?

**Comment by dalehenrich** [comment]
You can't just go around dropping support for perfectly viable platforms because it is inconvenient for the maintainers. If you insist on swiching to metadatless support for the older versions of Pharo, it should be possible to port metadataless support to those older platforms --- it should be relatively straight forward to code it up ... I do not have the cycles to do this personally, but I am willing to press the merge button when it comes time to do so ...

**Comment by dalehenrich** [comment]
... oh BTW, I am still relying on travis for applications that I've got running that use Pharo3.0 ---

**Comment by fniephaus** [comment]
Don't worry, I don't intend to deprecate P3 support anytime soon, there isn't a reason (yet). And I completely agree with what you've said. I'm running an experiment [here](https://travis-ci.org/hpi-swa/smalltalkCI/builds/298482976) to see what works and what not.
It'd be great if @estebanlm or someone else from the Pharo community could backport metadataless support, so we can update smalltalkCI and Metacello some time soon.

**Comment by fniephaus** [comment]
Ok, so P3 and P4 fail to load the metadataless format, `Squeak-4.5` and all other supported Squeak versions work fine (I believe they just always write out metadata, but that's a separate problem).

**Comment by dalehenrich** [comment]
That’s right ... tolerating metadataless format is much simpler to do ...

**Comment by dalehenrich** [comment]
Take a look at the difference between the implementations ... here's the GemStone version of [`hasMonticelloMetadata`]https://github.com/dalehenrich/filetree/blob/1b11ee48091bf20d26a8b289961fd770e9eb6752/repository/MonticelloFileTree-Core.package/MCFileTreeAbstractReader.class/instance/hasMonticelloMetadata.st) ... I found the [PR where I merged metadataless support for GemStone](https://github.com/dalehenrich/filetree/pull/198/files) ... more changes than I remember, but not giant amount of work either ...
